### PR TITLE
Add initial insertion mode errors

### PR DESCRIPTION
**Pull request by inikulin**
\cc @HTMLParseErrorWG/core

### CODE REVIEW

File: source [hunk 1]
```diff
@@ -100158,6 +100158,15 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
      rules don't apply any additional restrictions, make their way into the DOM.</p>
 
     <tr>
+     <td><dfn data-x="parse-error-non-conforming-doctype">non-conforming-doctype</dfn>
+     <td><p>This error occurs if the parser encounters the <span
+     data-x="syntax-doctype">DOCTYPE</span> that is correctly placed as a document preamble and that
```
Comment (pmdartus) [inline source]: data-x="syntax-doctype">DOCTYPE</span> ~that is~ correctly placed as a document preamble and ~that~

File: source [hunk 2]
```diff
@@ -100263,6 +100272,15 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
      </div>
 
     <tr>
+      <td><dfn data-x="parse-error-missing-doctype">missing-doctype</dfn>
+      <td>
+        <p>This error occurs if the parser encounters any content other than <span
+        data-x="syntax-comments">comments</span> or <span>ASCII whitespace</span> before <span
+        data-x="syntax-doctype">DOCTYPE</span> in the beginning of the document, i.e. DOCTYPE is not
```
Comment (pmdartus) [inline source]: data-x="syntax-doctype">DOCTYPE</span> ~in~ **at** the beginning of the document, i.e. DOCTYPE is not

**Review by pmdartus — DISMISSED** [review]
Some minor changes, except that LGTM.

**Comment by inikulin** [comment]
@pmdartus Thank you for the review. Fixed all nits, can you take yet another look and approve it if it's good to go, please?

**Review by pmdartus — APPROVED** [review]

**Pull request merged**